### ISSUE TITLE
Unable to generate Sequence Diagram: Error creating extended parser class

### ISSUE THREAD

**Issue by muirandy** [nickmcdowall/yatspec#11]
I'm seeing the following stacktrace at the end of my (single) test:

```
java.lang.RuntimeException: Error creating extended parser class: null

	at org.parboiled.Parboiled.createParser(Parboiled.java:58)
	at org.jtwig.parser.parboiled.ParserContext.instance(ParserContext.java:31)
	at org.jtwig.parser.parboiled.ParboiledJtwigParser.parse(ParboiledJtwigParser.java:37)
	at org.jtwig.parser.cache.InMemoryConcurrentPersistentTemplateCache.get(InMemoryConcurrentPersistentTemplateCache.java:39)
	at org.jtwig.parser.CachedJtwigParser.parse(CachedJtwigParser.java:19)
	at org.jtwig.JtwigTemplate.render(JtwigTemplate.java:98)
	at org.jtwig.JtwigTemplate.render(JtwigTemplate.java:74)
	at com.googlecode.yatspec.plugin.sequencediagram.PlantUmlMarkupGenerator.build(PlantUmlMarkupGenerator.java:42)
	at com.googlecode.yatspec.plugin.sequencediagram.PlantUmlMarkupGenerator.generateMarkup(PlantUmlMarkupGenerator.java:18)
	at com.googlecode.yatspec.plugin.sequencediagram.SequenceDiagramGenerator.generateSequenceDiagram(SequenceDiagramGenerator.java:29)
	at com.googlecode.yatspec.junit.SequenceDiagramExtension.lambda$afterTestExecution$0(SequenceDiagramExtension.java:55)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at com.googlecode.yatspec.junit.SequenceDiagramExtension.afterTestExecution(SequenceDiagramExtension.java:54)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAfterTestExecutionCallbacks$8(TestMethodTestDescriptor.java:229)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAllAfterMethodsOrCallbacks$12(TestMethodTestDescriptor.java:269)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAllAfterMethodsOrCallbacks$13(TestMethodTestDescriptor.java:269)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeAllAfterMethodsOrCallbacks(TestMethodTestDescriptor.java:268)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeAfterTestExecutionCallbacks(TestMethodTestDescriptor.java:228)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:133)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:71)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)
Caused by: java.lang.IllegalStateException
	at org.objectweb.asm.tree.analysis.BasicInterpreter.<init>(BasicInterpreter.java:66)
	at org.parboiled.transform.RuleMethodInterpreter.<init>(RuleMethodInterpreter.java:42)
	at org.parboiled.transform.InstructionGraphCreator.process(InstructionGraphCreator.java:41)
	at org.parboiled.transform.ParserTransformer.runMethodTransformers(ParserTransformer.java:62)
	at org.parboiled.transform.ParserTransformer.extendParserClass(ParserTransformer.java:45)
	at org.parboiled.transform.ParserTransformer.transformParser(ParserTransformer.java:38)
	at org.parboiled.Parboiled.createParser(Parboiled.java:54)
	... 63 more
```

I'm still getting the LSD page though without any sequence diagram.

Have you seen this before? Any clues?

**Comment by nickmcdowall** [comment]
Hey 👋. Thanks for the details. Please provide an example of your test so that I can try to reproduce the error.

**Comment by muirandy** [comment]
This does it for me:
```
@ExtendWith(SequenceDiagramExtension.class)
public class SimpleYatspecTest {
    private final TestState interactions = new TestState();

    @Test
    void yatspecLoggingTest() {
        interactions.log("Request from Client to Server", "hello");
    }
}
```

**Comment by muirandy** [comment]
I get these WARNINGS at the start of my test run if its relevant:

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.parboiled.transform.AsmUtils (file:/Users/andy/.m2/repository/org/parboiled/parboiled-java/1.1.7/parboiled-java-1.1.7.jar) to method java.lang.ClassLoader.findLoadedClass(java.lang.String)
WARNING: Please consider reporting this to the maintainers of org.parboiled.transform.AsmUtils
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

**Comment by muirandy** [comment]
A simple project with only the code above has no issues. There may be something going on with incompatibilities on dependencies or something.

**Comment by nickmcdowall** [comment]
Hey - just checked and I can see those warning too but doesn't cause the error. It's a strange one - I would need to bring in the same dependencies to try and reproduce. Can you confirm Java version and Junit engine etc?

**Comment by muirandy** [comment]
I have a project using Quarkus 1.9.2.Final. This has its own tweeked version of Junit5. I don't understand all the ins-and-outs to know what they do to it. I suspect there's something thats not compatible. Not an easy one.

**Comment by muirandy** [comment]
Here's the project: https://github.com/muirandy/distributed-yatspec-example
SimpleYatspecTest is the best place to look (https://github.com/muirandy/distributed-yatspec-example/blob/master/src/test/java/com/github/muirandy/docs/yatspec/distributed/example/SimpleYatspecTest.java).

If you decide to build it, you can get rid of all the other code under src. then remove the com.github.muirandy dependencies, replacing with your Yatspec.

**Comment by nickmcdowall** [comment]
Hi @muirandy - I finally got around to trying out the project you mentioned above. It looks to be working for me on the latest version. (I have removed quite a few of the dependencies that used to be required in older versions of the yatspec library so maybe that has helped 🤷 ).

```
        <dependency>
            <groupId>com.github.nickmcdowall</groupId>
            <artifactId>yatspec</artifactId>
            <version>2020.3.12</version>
        </dependency>
```

![image](https://user-images.githubusercontent.com/1330362/103948615-f9bb9680-5131-11eb-8444-d8bcf644c9c9.png)
